internal/service/route53resolver/firewall_config_data_source.go

```diff
@@ -0,0 +1,60 @@
+package route53resolver
+
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/route53resolver"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-provider-aws/internal/conns"
+)
+
+func DataSourceResolverFirewallConfig() *schema.Resource {
```

Comment:
Datasource name should match corresponding resource (ResourceFirewallConfig).

```suggestion
func DataSourceFirewallConfig() *schema.Resource {
```